src/app/[lng]/[partnerTheme]/layout.tsx

```diff
@@ -1,27 +1,45 @@
 import { getPartnerThemes } from '@/app/lib/getPartnerThemes';
 import { FeatureCards } from '@/components/FeatureCards';
+import { partnerThemeSchema } from '@/utils/validation-schemas';
 import type { Metadata } from 'next';
 import { notFound } from 'next/navigation';
 import React from 'react';
-import { Layout } from 'src/Layout';
+import { Layout as BaseLayout } from 'src/Layout';
 
 export async function generateMetadata({
   params: { partnerTheme },
 }: {
   params: { partnerTheme: string };
 }): Promise<Metadata> {
-  const partnerThemes = await getPartnerThemes();
-  const partnerThemesData = partnerThemes.data?.find(
-    (d) => d.attributes?.uid === partnerTheme,
-  );
-  const theme = partnerThemesData
-    ? partnerThemesData.attributes?.uid
-    : 'default';
-  return {
-    other: {
-      'partner-theme': theme,
-    },
-  };
+  try {
+    // Validate partner theme format
+    const result = partnerThemeSchema.safeParse(partnerTheme);
+    if (!result.success) {
+      throw new Error('Invalid partner theme format');
+    }
+
+    // Check if partner theme exists in Strapi
+    const partnerThemes = await getPartnerThemes();
+    const partnerThemesData = partnerThemes.data?.find(
+      (d) => d.attributes?.uid === result.data,
+    );
+
+    if (!partnerThemesData) {
```

Comment:
Behavior has changed compare to before, that is not widely supported by the app and will crash it, please do not do unintended changes